docs/specs/authentication-flow-prioritized-identity.md

```diff
@@ -0,0 +1,268 @@
+# Authentication Flow - Prioritized Identity
+
+- [Introduction](#introduction)
+- [Usecases](#usecases)
+  - [Force user to use a preferred authentication method if available](#force-user-to-use-a-preferred-authentication-method-if-available)
+- [Behaviors](#behaviors)
+  - [When there are multiple levels of priority](#when-there-are-multiple-levels-of-priority)
+  - [When the user has only some of the prioritized identities](#when-the-user-has-only-some-of-the-prioritized-identities)
+## Introduction
+
+It is common that some identities are considered more secure, or more preferable comparing with other identities of the same user. Apps might want to have a way to specify the priority of different types of identities in a login flow, so when a user has another identity which is more preferable, it will be used for identification inside a login flow.
+
+## Usecases
+
+### Force user to use a preferred authentication method if available
```

Comment:
BTW @tung2744 this is not a "use case", you're literally just writing out what the feature want to do instead of mentioning a use case.

Use cases could be:
- Force users to use a safer OAuth Provider (such as Google Login)
- Migrating users from email identities to phone identities
- Internal ADFS always have priority over email login for security purpose

Than in each use case it would be easier to describe what's needed exactly, and easier for us to consider what might be missing in the config.

Reply:
cc: @louischan-oursky 

(p.s. I paid special attention on it this time since Tung is new to it)

Reply:
Then I think the use case here should be "Force users to use a safer OAuth Provider (such as Google Login)". Let me update this section.

> Migrating users from email identities to phone identities

However I am not sure if this usecase should be handled by this feature. Maybe I will not include it in the spec at the moment.